### PR TITLE
adjustments

### PR DESCRIPTION
**Pull request by rjvillanueva89**
- updated packages with -D "@typescript-eslint/eslint-plugin" for Error: Definition for rule '@typescript-eslint/no-unused-vars' was not found
- added .eslintrc.js to root directories to apply some rules

added to b2c:
- components/ApolloClientProvider
- components/HasuraAdapter
- components/PageAuth
- pages/auth
- pages/auth/login
- api/auth/[...nextauth]

added to packages/ui:
- Icons
- Layouts/Panel

added to packages/utils:
- doSSRFetch
- inArray

### CODE REVIEW

**Review by jubalm — CHANGES_REQUESTED** [review]
This is a huge PR and I have very limited time at the moment. Leaving some partial notes as I have more time to review

File: apps/b2c/pages/_app.tsx [hunk 1]
```diff
@@ -1,8 +1,37 @@
-import '../styles/globals.css'
+import ApolloClientProvider from 'components/ApolloClientProvider'
+import PageAuth from 'components/PageAuth'
+import { SessionProvider } from 'next-auth/react'
 import type { AppProps } from 'next/app'
+import Head from 'next/head'
+import '../styles/globals.css'
+
+function App({ Component, pageProps: { session, ...pageProps } }: AppProps) {
+  const allowedRoles = (Component as any).allowedRoles
```
[the remark this answers is not in the document]
Comment (pgdureza) [reply apps/b2c/pages/_app.tsx:9]: agreed. can create a specific type for consumer pages if this is the intent. 

```
export type ConsumerPage = NextPage & {
  getLayout?: React.FC<PropsWithChildren>
  allowedRoles?: Role[]
}
```


on the app component itself, can just create a new interface that overrides Component.
```
interface ConsumerApp extends AppProps {
  Component: ConsumerPage
}
```

File: apps/b2c/pages/api/auth/[...nextauth].ts [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+        })
+        return decodedToken as Awaitable<JWT>
+      },
+    },
+    callbacks: {
+      jwt: async ({ token, user }) => {
+        if (user) {
+          const role = user.role ? user.role : 'user'
+
+          token = {
```
Comment (jubalm) [inline apps/b2c/pages/api/auth/[...nextauth].ts:89]: can this be written in a more immutable manner?
```ts
const jwtResponse = {
  ...token,
  // rest
}
```

File: apps/b2c/pages/api/auth/[...nextauth].ts [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+        }
+
+        return Promise.resolve(token)
+      },
+      session: async ({ session, token }) => {
+        const encodedToken = jwt.sign(token, process.env.NEXTAUTH_SECRET!, {
+          algorithm: 'HS256',
+        })
+
+        session.user = {
```
Comment (jubalm) [inline apps/b2c/pages/api/auth/[...nextauth].ts:109]: can be more immutable ex:
```ts
const sessionResponse: User = {
  user: { //.. },
  token: { //.. }
}
```

not familiar with the api but it's nice to check if `callbacks.session` is already strictly typed and use those for inference rather that using `as User`

File: apps/b2c/pages/api/auth/[...nextauth].ts [hunk 4]
```diff
@@ -0,0 +1,155 @@
...
+    },
+  })
+}
+
+interface GetExpProps {
+  exp?: number
+  maxAge?: number
+}
+
+const getExp = ({ exp, maxAge = MAX_AGE }: GetExpProps) => {
```
Comment (jubalm) [inline apps/b2c/pages/api/auth/[...nextauth].ts:138]: can we give this a clearer name? like `getExpiry`?

File: apps/b2c/pages/api/auth/[...nextauth].ts [hunk 5]
```diff
@@ -0,0 +1,155 @@
...
+  const now = Math.floor(Date.now() / 1000)
```
Comment (jubalm) [inline apps/b2c/pages/api/auth/[...nextauth].ts:139]: not sure if there's merit in converting this to seconds when you only want to try to compare two dates?

File: apps/b2c/pages/auth/index.tsx [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+export const REDIRECT_MAP: Record<Role, string> = {
+  user: '/',
+  public: '/',
+}
+
+const Auth = () => {
+  const { push } = useRouter()
+  const { status, data } = useSession()
+
+  if (status === 'loading') return null
```
Comment (jubalm) [inline apps/b2c/pages/auth/index.tsx:13]: are we planning on adding a 'loading' state here in the future? we can use the issues page to drop plans and discussions

File: packages/utils/inArray.ts [hunk 7]
```diff
@@ -0,0 +1,5 @@
+const inArray = <T>(needle: T, haystack: T[]): boolean => {
```
[the remark this answers is not in the document]
Comment (rjvillanueva89) [reply packages/utils/inArray.ts]: agreed 😅

File: apps/b2c/pages/api/auth/[...nextauth].ts (hunk 3 above)
Comment (rjvillanueva89) [reply apps/b2c/pages/api/auth/[...nextauth].ts]: updated

File: apps/b2c/pages/api/auth/[...nextauth].ts (hunk 2 above)
Comment (rjvillanueva89) [reply apps/b2c/pages/api/auth/[...nextauth].ts]: updated

File: apps/b2c/pages/api/auth/[...nextauth].ts (hunk 4 above)
Comment (rjvillanueva89) [reply apps/b2c/pages/api/auth/[...nextauth].ts]: updated

File: apps/b2c/pages/api/auth/[...nextauth].ts (hunk 5 above)
Comment (rjvillanueva89) [reply apps/b2c/pages/api/auth/[...nextauth].ts:141]: I'm currently using the function to
- set the initial exp; if exp is missing
- extend the expiration (if the user reloads the site and token isn't expired)

`
const getExpiry = ({ exp, maxAge = MAX_AGE }: GetExpiryProps) => {
  const now = Math.round(Date.now() / 1000)

  // set initial expiration
  if (!exp) return now + maxAge

  // check exp if isn't expired ? extend : exp
  return exp && exp > now ? now + maxAge : exp
}
`

need to verify if the extend is working

Comment (jubalm) [reply apps/b2c/pages/api/auth/[...nextauth].ts:141]: My confusion is why the need to divide date values by 1000 (presumably to convert to seconds) when the operation is date comparison? i suspect the env is not a standard date value? but then again, whats preventing it from being one?

Comment (rjvillanueva89) [reply apps/b2c/pages/api/auth/[...nextauth].ts:141]: I should rename this to setExpiration; because that's the main purpose of the function.

**Review by pgdureza — APPROVED** [review]

**Review by jubalm — APPROVED** [review]

**Pull request merged**